core/org.wso2.carbon.bootstrap/src/test/java/org/wso2/carbon/bootstrap/CarbonSecurityManagerTest.java

```diff
@@ -0,0 +1,113 @@
+    private CarbonSecurityManager carbonSecurityManager;
+
+    /**
+     * Test if creation of an instance of CarbonSecurityManager throws
+     * IllegalArgumentException when denied.system.properties is not specified
+     */
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testCarbonSecurityManager() {
+        carbonSecurityManager = new CarbonSecurityManager();
+    }
+
+    /**
+     * Test printDeniedSystemProperties method when denied.system.properties is specified
+     * The method is private and accessed by the constructor
+     */
+    @Test
+    public void testPrintDeniedSystemProperties() {
+        System.setProperty("denied.system.properties", "mockDeniedProperty");
+        carbonSecurityManager = new CarbonSecurityManager();
+    }
+
+    /**
+     * Test checkPropertyAccess method for an allowed property using a mock carbonSecurityManager
+     */
+    @Test
+    public void testCheckPropertyAccess() {
+        carbonSecurityManager = mock(CarbonSecurityManager.class);
+        carbonSecurityManager.checkPropertiesAccess();
+        Mockito.verify(carbonSecurityManager).checkPropertiesAccess();
+    }
+
+    /**
+     * Test testCheckPropertyAccess throws AccessControlException for a denied property
+     */
+    @Test(expectedExceptions = AccessControlException.class)
+    public void testCheckPropertyAccessDeniedProperty() {
+        String key = "mockDeniedProperty";
+        System.setProperty("denied.system.properties", "mockDeniedProperty");
+        carbonSecurityManager = new CarbonSecurityManager();
+        carbonSecurityManager.checkPropertyAccess(key);
+    }
+
+    /**
+     * Test if checkAccessThread method checks permission for a Thread
+     */
+    @Test
+    public void testCheckAccessThread() {
+        carbonSecurityManager = mock(CarbonSecurityManager.class);
+        Thread thread = mock(Thread.class);
+        carbonSecurityManager.checkAccess(thread);
+    }
+
+    /**
+     * Test if checkAccessThread method checks permission for a ThreadGroup
+     */
+    @Test
+    public void testCheckAccessThreadGroup() {
+        carbonSecurityManager = mock(CarbonSecurityManager.class);
+        ThreadGroup threadGroup = mock(ThreadGroup.class);
+        carbonSecurityManager.checkAccess(threadGroup);
+    }
+
+    /**
+     * Test if checkAccessThread handles a null Thread
+     */
+    @Test
+    public void testCheckAccessNullThread() {
+        carbonSecurityManager = mock(CarbonSecurityManager.class);
+        Thread thread = null;
+        assertNull(thread);
+        carbonSecurityManager.checkAccess(thread);
+    }
+
+    /**
+     * Test if checkAccessThread handles a null ThreadGroup
+     */
+    @Test
+    public void testCheckAccessNullThreadGroup() {
+        carbonSecurityManager = mock(CarbonSecurityManager.class);
+        ThreadGroup threadGroup = null;
+        assertNull(threadGroup);
+        carbonSecurityManager.checkAccess(threadGroup);
```

Comment:
checkAccess method with a null value should throw an exception. IMO, we do not need to mock this class. This method should be altered to test for a null pointer expected exception.

Reply:
Fixed in cc1487d027afcdd325678ef87fbc5f77607b7171